org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -299,7 +299,7 @@ protected void addProjectSourceFolders(IClasspathDescriptor classpath, Map<Strin
       String mainResourcesEncoding = null;
       String testResourcesEncoding = null;
 
-      boolean isTestCompilationSkipped = false;
+      boolean isTestCompilationSkipped = true;
       boolean isTestResourcesSkipped = false;
```

Comment:
You probably want to do the same for resources, don't you?

Reply:
Yes, I added the code to the PR